src/main/java/frc/robot/SN_SwerveModule.java

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFXConfiguration;
+import com.ctre.phoenix.sensors.CANCoder;
+import com.frcteam3255.components.motors.SN_TalonFX;
+import com.frcteam3255.utils.CTREModuleState;
+import com.frcteam3255.utils.SN_Math;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.math.util.Units;
+import frc.robot.RobotPreferences.prefDrivetrain;
+
+/**
+ * Coaxial swerve module with SN_TalonFX's for both driving and steering. Uses
+ * CANCoder's for the absolute steering encoders.
+ */
+public class SN_SwerveModule {
+
+  public int moduleNumber;
+
+  private SN_TalonFX driveMotor;
+  private SN_TalonFX steerMotor;
+
+  private CANCoder absoluteEncoder;
+  private double absoluteEncoderOffset;
+
+  private TalonFXConfiguration driveConfiguration;
+  private TalonFXConfiguration steerConfiguration;
+
+  private double lastAngle;
+
+  /**
+   * Create a new SN_SwerveModule.
+   * 
+   * @param moduleConstants Constants required to create a swerve module
+   */
+  public SN_SwerveModule(SN_SwerveModuleConstants moduleConstants) {
+    moduleNumber = moduleConstants.number;
+
+    driveMotor = new SN_TalonFX(moduleConstants.driveMotorID);
+    steerMotor = new SN_TalonFX(moduleConstants.steerMotorID);
+
+    absoluteEncoder = new CANCoder(moduleConstants.absoluteEncoderID);
+    absoluteEncoderOffset = moduleConstants.absoluteEncoderOffset;
+
+    driveConfiguration = new TalonFXConfiguration();
+    steerConfiguration = new TalonFXConfiguration();
+
+    lastAngle = 0;
+
+    configure();
+  }
+
+  /**
+   * Configure the drive motor, steer motor, and absolute encoder.
+   */
+  public void configure() {
+    // driveMotor
+
+    driveMotor.configFactoryDefault();
```

Comment:
fun fact. SN_TalonFX does this in it's constructor

Reply:
cool, still gonna do it here though